### ISSUE TITLE
[Feature Request]: Improvement on DJ Hero Turntable Emulation

### ISSUE THREAD

**Issue by KingS1zzle** [PCSX2/pcsx2#9775]
### Description

The current build of the DJ Hero Turntable Controller Emulation was implemented by [sanjay900](https://github.com/sanjay900) and is functional with a few bugs. 

- Always Scratching/Spinning in one direction
- When Buttons are tapped, the game thinks you are trying to hold and Scratch/Spin.
- Effects Knob acts like a Steering wheel. When rotated, it rotates back to 0.

These bugs happen if you use a Xbox/PS Controller or a DJ Hero Turntable Controller. 

On a Xbox 360 DJ Hero Turntable Controller:
- Turntable Axis isn't being detected at all.

### **In Reference to** 
#9129 

### Reason

DJ Hero is playable, but multiple notes in the game arn't triggering, including functions and specials.

- Playing above normal is much harder since Up/Down Scratches are only detectable if the Turntable Emulation if you have previously set the turntable direction. Which there are multiple charts that require you to go back and forth.
- Achieving a high combo seems impossible since Scratching/Spinning is always on, which the game detects tapping notes as a scratch, losing your combo if you had one.
- You need a 50 note combo to earn a Rewind, rewinding lets you replay parts of the song and earning you more score.
- If you earn a Rewind, Id assume you would use it instantly since it requires you to spin the record backwards.
- Multiple parts of the song can give you Euphoria/Star Power, you need to full combo to earn a bar. Since the notes are half broken, seems like luck when you earn a bar.

### Examples

Effects Knob could work like: Turning left/right and once the player stops turning, stay at the current rotation.

Maybe there could be an added setting that disables the auto Scratching/Spinning so normal scratching could be used on Turntable controllers.
Auto Scratching is still useful for Controllers, but breaks the game with tapping and Up/Down notes.

**Comment by sanjay900** [comment]
Yeah i didn't really care much about it working with a controller when i wrote this, it was strictly for actual turntables.
For a 360 turntable, the default mappings are based on that, since actual ps3 turntables won't work with this as they don't expose the necessary inputs in their HID descriptors for a PC to recognize the inputs. There are apps out there though for emulating a 360 turntable when using a ps3 turntable.

The default bindings should work just find for a 360 turntable, but you will need to change the multiplier to get it to recognise it, as the 360 turntables don't use much of the axis' range, presumably because the creators didn't want the turntable controlling the 360 menus on an actual console. I suspect this is also the same reason PCSX2 isn't detecting it - there must be some deadzone handling ignoring the axis somewhere.

**Comment by sanjay900** [comment]
I actually suspect that the deadzone is implemented somewhere on the SDL side, which is unfortunate as i don't really see an easy way to fix that. If i for example change my XB360 turntable to use a higher range in hardware, it works perfectly fine.

**Comment by sanjay900** [comment]
One thing i did note - on windows, you can calibrate the joystick using the windows `joy.cpl` tool, and then use DInput, which *does* actually respect your calibration. Both SDL and XInput do *not* appear to do this however, they just pass the value across raw.

**Comment by dAn84imola** [comment]
Hy @sanjay900 and @KingS1zzle I join this dated conversation in order to give some (hope) useful feedback. First of all, I'm not a skilled user of Linux or in general speaking of software so I apologize if something written by me is obvious.
 
I recently tested my EU turntable on the well know Batocera (stable V38 X86-64) which is implemented with the pcsx2 app inside and can confirm all the issues above but in other way. Batocera does mirror the hardware as _"redoctante dj guitar hero 5 ps3"_ through his original usb dongle and buttons with digital cross arrows are well remapped under EmulationStation. Now:

1- In game there is no activity of the turntable except start+select to close the game (shortcut for EmulationStation to main menu).
2- Inside the app of pcsx2 all buttons are correctly recognized and remapped (automapping does not work) except the crossfader and the knob duo no input/activity recognized.

What describes is true on pcsx2, but on pcsx2+ all buttons work in game except the turntable, the crossfader, and the knob.
I attach some picture to be more clarify. Thank you for the patience and going further if possible to make functional the turntable.

![IMG_20240313_134049](https://github.com/PCSX2/pcsx2/assets/127186401/86248da1-550e-4158-bbb7-5c4d09d5a420)
![IMG_20240313_140833](https://github.com/PCSX2/pcsx2/assets/127186401/b3d01d16-c2ee-4c45-ae21-481a7736be5a)
![IMG_20240313_190543](https://github.com/PCSX2/pcsx2/assets/127186401/2c876b84-52ef-40b1-855c-b29a5a6ba0d7)
![IMG_20240313_140636](https://github.com/PCSX2/pcsx2/assets/127186401/96b3b7ff-370b-4127-b52b-1a5fcc83fb04)

**Comment by sanjay900** [comment]
The PS3 turntable does not expose inputs in a way that the emulator can recognise. 

For Linux compatibility, we will need to patch the kernel to expose the hidden axis, much like I did a while ago for guitars and tilt. I haven't really had the time to do this, but at some point I probably will get around to it.

The default mappings are designed for the Xbox 360 turntable, as that is the turntable that will work most easily on windows, so that is any none of the auto mappings work.
